src/Ast.ml

```diff
@@ -60,10 +60,22 @@ let rec is_sugared_list exp =
   | _ -> false
 
 
-let rec is_trivial exp =
+let would_force_break (c: Conf.t) s =
+  let contains_internal_newline s =
+    match String.rindex s '\n' with
+    | None -> false
+    | Some i when i = String.length s - 1 -> false
+    | _ -> true
+  in
+  Poly.equal c.break_string_literals `Newlines
+  && contains_internal_newline s
+
+
+let rec is_trivial c exp =
   match exp.pexp_desc with
+  | Pexp_constant Pconst_string (s, None) -> not (would_force_break c s)
```

Comment:
To prevent string literals that will force a line break from being considered 'trivial', and thereby 'simple', which the application argument formatting is not prepared to handle.